api/core/database/models/api/bot/reminder.py

```diff
@@ -0,0 +1,53 @@
+from typing import NoReturn, Union
+
+from sqlalchemy import ARRAY, BigInteger, Boolean, Column, DateTime, ForeignKey, Integer, String
+from sqlalchemy.orm import relationship
+from sqlalchemy.orm import validates
+
+from api.core.database import Base
+
+
+class Reminder(Base):
+    """A reminder created by a user."""
+
+    __tablename__ = 'reminder'
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+
+    # Whether this reminder is still active.
+    # If not, it has been sent out to the user.
+    active = Column(Boolean, nullable=False)
+    # The channel ID that this message was
+    # sent in, taken from Discord.
+    channel_id = Column(BigInteger, nullable=False)
+
+    # The content that the user wants to be reminded of.
+    content = Column(String(1500), nullable=False)
+
+    # When this reminder should be sent.
+    expiration = Column(DateTime(True), nullable=False)
+
+    author_id = Column(ForeignKey('user.id', deferrable=True, initially='DEFERRED'), nullable=False, index=True)
+
+    # The jump url to the message that created the reminder
+    jump_url = Column(String(88), nullable=False)
+    # IDs of roles or users to ping with the reminder.
+    mentions = Column(ARRAY(BigInteger()), nullable=False, default=[])
+
+    # The creator of this reminder.
+    author = relationship('User', cascade="all, delete")
+
+    @validates('channel_id')
+    def validate_rchannel_id(self, _key: str, channel_id: int) -> Union[int, NoReturn]:
```

Comment:
```suggestion
    def validate_channel_id(self, _key: str, channel_id: int) -> Union[int, NoReturn]:
```
Small typo.

Reply:
No, actually it wasn't.  [See Offensive Message](https://github.com/D0rs4n/api/blob/827e4645c3213c0be37e217ff38bf6994619ae6a/api/core/database/models/api/bot/offensive_message.py#L32)
However, a few of the function names need to be renamed to keep things consistent, so it's a valid concern. Thank you.